hw/bsp/family_support.cmake

```diff
@@ -367,7 +373,7 @@ function(family_flash_jlink TARGET)
   endif ()
 
   file(GENERATE
-    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${TARGET}.jlink
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>/${TARGET}.jlink
```

Comment:
does IAR has 1 config per folder within the build dir. Normally clion has 1 build dir per config. If possible, change it to `{TARGET}_$<CONFIG>.jlink` would be better.

Reply:
Yes each binary has it's own folder.
```
$ ls
audio_4_channel_mic_Debug.jlink           build.ninja          build-RelWithDebInfo.ninja  CMakeFiles/  RelWithDebInfo/
audio_4_channel_mic_Release.jlink         build-Debug.ninja    cmake_install.cmake         Debug/       tinyusb/
audio_4_channel_mic_RelWithDebInfo.jlink  build-Release.ninja  CMakeCache.txt              Release/
```

Reply:
I see, for clion, I typically use one build dir per config. I think maybe we can use 
$<TARGET_FILE_DIR:${TARGET}> instead of of ${CMAKE_CURRENT_BINARY_DIR} for the flash target

```
$ ls
audio_test.elf  audio_test.elf.map  audio_test.jlink  CMakeFiles  cmake_install.cmake  tinyusb
``

Reply:
It gives something like this:
```
.
├── build.ninja
├── build-Debug.ninja
├── build-Release.ninja
├── build-RelWithDebInfo.ninja
├── cmake_install.cmake
├── CMakeCache.txt
├── CMakeFiles
│   └──...
├── Debug
│   ├── audio_4_channel_mic.elf
│   ├── audio_4_channel_mic.elf.map
│   ├── audio_4_channel_mic.jlink
│   └── libboard_stm32f439nucleo.a
├── Release
│   ├── audio_4_channel_mic.elf
│   ├── audio_4_channel_mic.elf.map
│   ├── audio_4_channel_mic.jlink
│   └── libboard_stm32f439nucleo.a
├── RelWithDebInfo
│   ├── audio_4_channel_mic.elf
│   ├── audio_4_channel_mic.elf.map
│   ├── audio_4_channel_mic.jlink
│   └── libboard_stm32f439nucleo.a
└── tinyusb
    ├── cmake_install.cmake
    ├── CMakeFiles
    │   └──...
    ├── Debug
    │   └── libaudio_4_channel_mic-tinyusb.a
    ├── Release
    │   └── libaudio_4_channel_mic-tinyusb.a
    └── RelWithDebInfo
        └── libaudio_4_channel_mic-tinyusb.a
```

Reply:
perfect, it is what we want the .jlink file in the same folder of the elf for both IAR Workbecnh and clion. Not sure if IAR can execute the -jlink target, if it does, it should be able to flash your board with correct elf

Reply:
Done.

Currently CMake support is pretty primitive, it defaults to Ninja multi config generator and there is no option for pre/post build actions. And I had multiple crashes modifying CMake file.

I'm going to give them some feedbacks.